api/v1alpha4/nutanixcluster_conversion.go

```diff
@@ -19,76 +19,72 @@ package v1alpha4
 import (
 	infrav1beta1 "github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1"
 	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	//nolint:staticcheck // sigs.k8s.io/cluster-api/api/v1alpha4 is deprecated and will be removed in one of the next releases.
 	capiv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
 	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 // ConvertTo converts this NutanixCluster to the Hub version (v1beta1).
-func (src *NutanixCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
+func (src *NutanixCluster) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1beta1.NutanixCluster)
 	return Convert_v1alpha4_NutanixCluster_To_v1beta1_NutanixCluster(src, dst, nil)
 }
 
 // ConvertFrom converts from the Hub version (v1beta1) to this NutanixCluster.
-func (dst *NutanixCluster) ConvertFrom(srcRaw conversion.Hub) error { //nolint
+func (dst *NutanixCluster) ConvertFrom(srcRaw conversion.Hub) error {
 	src := srcRaw.(*infrav1beta1.NutanixCluster)
 	return Convert_v1beta1_NutanixCluster_To_v1alpha4_NutanixCluster(src, dst, nil)
 }
 
 // ConvertTo converts this NutanixClusterList to the Hub version (v1beta1).
-func (src *NutanixClusterList) ConvertTo(dstRaw conversion.Hub) error { //nolint
+func (src *NutanixClusterList) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1beta1.NutanixClusterList)
 	return Convert_v1alpha4_NutanixClusterList_To_v1beta1_NutanixClusterList(src, dst, nil)
 }
 
 // ConvertFrom converts from the Hub version(v1beta1) to this NutanixClusterList.
-func (dst *NutanixClusterList) ConvertFrom(srcRaw conversion.Hub) error { //nolint
+func (dst *NutanixClusterList) ConvertFrom(srcRaw conversion.Hub) error {
 	src := srcRaw.(*infrav1beta1.NutanixClusterList)
 	return Convert_v1beta1_NutanixClusterList_To_v1alpha4_NutanixClusterList(src, dst, nil)
 }
 
 // Convert_v1alpha4_NutanixClusterSpec_To_v1beta1_NutanixClusterSpec converts NutanixClusterSpec in NutanixClusterResource from v1alpha4 to v1beta1 version.
-//
-//nolint:all
```

Comment:
A lot of `nolint` directives have been removed and new ones, more precise ones, added. This is because some nolint directives were present for no reason and could just be removed while some nolint directives were to broad and highlighted as problematic by the nolintlint linter.